src/components/Comment/index.spec.tsx

```diff
@@ -24,22 +24,40 @@ describe(__filename, () => {
     Partial<DefaultProps>;
 
   const render = ({
-    store = configureStore(),
+    beginEdit = false,
     commentId = null,
+    fileName = null,
+    line = null,
+    store = configureStore(),
+    versionId = nextUniqueId(),
     ...moreProps
-  }: RenderParams = {}) => {
-    const props = {
-      addonId: 1,
+  }: RenderParams & { beginEdit?: boolean } = {}) => {
+    const keyParams = {
       commentId,
-      fileName: null,
-      line: null,
-      readOnly: false,
-      versionId: 2,
+      fileName,
+      line,
+      versionId,
+    };
+    if (beginEdit) {
+      store.dispatch(
+        actions.beginComment({
+          ...keyParams,
+          commentId: commentId || undefined,
+        }),
+      );
+    }
+    const props = {
+      addonId: nextUniqueId(),
+      ...keyParams,
       ...moreProps,
     };
-    return shallowUntilTarget(<Comment {...props} />, CommentBase, {
+    const dispatchSpy = spyOn(store, 'dispatch');
+
+    const root = shallowUntilTarget(<Comment {...props} />, CommentBase, {
       shallowOptions: { context: { store } },
     });
+
+    return { root, dispatchSpy };
```

Comment:
In other tests, where we sometimes need something in addition to `root` returned from a `render` function, we tend to have a base `render` which just returns `root`, as well as an extra function, which returns `root` plus something else, e.g., [`renderWithInstance`](https://github.com/mozilla/addons-code-manager/blob/5444b4e9b4deb45881538c58f56841fd8ea5ef0b/src/components/CodeOverview/index.spec.tsx#L66). I wonder if we can take a similar approach here, having a base `render` and a `renderWithDispatch`? We could then just call `render` when we don't need access to `dispatch` in the test, which seems to be the case in a number of tests.

I think this would reduce the size of the diff as well.